src/System.Data.OleDb/src/ColumnBinding.cs

```diff
@@ -0,0 +1,1500 @@
+
+            _ordinal = binding.iOrdinal.ToInt32();
+            _maxLen = binding.cbMaxLen.ToInt32();
+            _wType = binding.wType;
+            _precision = binding.bPrecision;
+
+            _ifIRowsetElseIRow = ifIRowsetElseIRow;
+
+            SetSize(Bindings.ParamSize.ToInt32());
+        }
+
+        internal Bindings Bindings {
+            get {
+                _bindings.CurrentIndex = IndexWithinAccessor;
+                return _bindings;
+            }
+        }
+
+        internal RowBinding RowBinding {
+             get { return _rowbinding; }
+        }
+
+        internal int ColumnBindingOrdinal {
+            get { return _ordinal; }
+        }
+
+        private int ColumnBindingMaxLen {
+            get { return _maxLen; }
+        }
+
+        private byte ColumnBindingPrecision {
+            get { return _precision; }
+        }
+
+        private short DbType {
+            get { return _wType; }
+        }
+
+        private Type ExpectedType {
+            get { return NativeDBType.FromDBType(DbType, false, false).dataType; }
+        }
+
+        internal int Index {
+            get { return _index; }
+        }
+        internal int IndexForAccessor {
+            get { return _indexForAccessor; }
+        }
+
+        internal int IndexWithinAccessor {
+            get { return _indexWithinAccessor; }
+        }
+
+        private int ValueBindingOffset { // offset within the value of where to start copying
+            get { return _valueBindingOffset; }
+        }
+
+        private int ValueBindingSize { // maximum size of the value to copy
+            get { return _valueBindingSize; }
+        }
+
+        internal int ValueOffset { // offset within the native buffer to put the value
+            get { return _offsetValue; }
+        }
+
+        private OleDbDataReader DataReader() {
+            Debug.Assert(null != _dataReader, "null DataReader");
+            return _dataReader;
+        }
+
+        internal bool IsParameterBindingInvalid(OleDbParameter parameter) {
+            Debug.Assert((null != _parameter) && (null != parameter), "null parameter");
+            return ((_parameter.ChangeID != _parameterChangeID) || (_parameter != parameter));
+        }
+
+        internal bool IsValueNull() {
+            return ((DBStatus.S_ISNULL == StatusValue())
+                    || (((NativeDBType.VARIANT == DbType) || (NativeDBType.PROPVARIANT == DbType))
+                        && (Convert.IsDBNull(ValueVariant()))));
+        }
+
+        private int LengthValue() {
+            int length;
+            if (_ifIRowsetElseIRow) {
+                length = RowBinding.ReadIntPtr(_offsetLength).ToInt32();
+            }
+            else { // WebData 94427
+                length = Bindings.DBColumnAccess[IndexWithinAccessor].cbDataLen.ToInt32();
+            }
+            return Math.Max(length, 0);
+        }
+        private void LengthValue(int value) {
```

Comment:
Can this be put through the code formatter to take care of absence of newlines.

Reply:
My guess was that @maryamariyan was going to do code formatting as the very last step after getting all tests passing. That's because if there's a bug to track down, it's helpful to be able to diff the code with the original.